src/manybody_allegro.F

```diff
@@ -0,0 +1,480 @@
+                                              sp
+   USE pair_potential_types,            ONLY: allegro_pot_type,&
+                                              allegro_type,&
+                                              pair_potential_pp_type,&
+                                              pair_potential_single_type
+   USE particle_types,                  ONLY: particle_type
+   USE torch_api,                       ONLY: torch_dict_create,&
+                                              torch_dict_get,&
+                                              torch_dict_insert,&
+                                              torch_dict_release,&
+                                              torch_dict_type,&
+                                              torch_model_eval,&
+                                              torch_model_load
+   USE util,                            ONLY: sort
+#include "./base/base_uses.f90"
+
+   IMPLICIT NONE
+
+   PRIVATE
+   PUBLIC :: setup_allegro_arrays, destroy_allegro_arrays, &
+             allegro_energy_store_force_virial, allegro_add_force_virial
+   CHARACTER(len=*), PARAMETER, PRIVATE :: moduleN = 'manybody_allegro'
+
+CONTAINS
+
+! **************************************************************************************************
+!> \brief ...
+!> \param nonbonded ...
+!> \param potparm ...
+!> \param glob_loc_list ...
+!> \param glob_cell_v ...
+!> \param glob_loc_list_a ...
+!> \param unique_list_a ...
+!> \param cell ...
+!> \par History
+!>      Implementation of the allegro potential - [gtocci] 2023
+!> \author Gabriele Tocci - University of Zurich
+! **************************************************************************************************
+   SUBROUTINE setup_allegro_arrays(nonbonded, potparm, glob_loc_list, glob_cell_v, glob_loc_list_a, &
+                                   unique_list_a, cell)
+      TYPE(fist_neighbor_type), POINTER                  :: nonbonded
+      TYPE(pair_potential_pp_type), POINTER              :: potparm
+      INTEGER, DIMENSION(:, :), POINTER                  :: glob_loc_list
+      REAL(KIND=dp), DIMENSION(:, :), POINTER            :: glob_cell_v
+      INTEGER, DIMENSION(:), POINTER                     :: glob_loc_list_a, unique_list_a
+      TYPE(cell_type), POINTER                           :: cell
+
+      CHARACTER(LEN=*), PARAMETER :: routineN = 'setup_allegro_arrays'
+
+      INTEGER                                            :: handle, i, iend, igrp, ikind, ilist, &
+                                                            ipair, istart, jkind, nkinds, nlocal, &
+                                                            npairs, npairs_tot
+      INTEGER, DIMENSION(:), POINTER                     :: temp_unique_list_a, work_list, work_list2
+      INTEGER, DIMENSION(:, :), POINTER                  :: list
+      REAL(KIND=dp), DIMENSION(3)                        :: cell_v, cvi
+      REAL(KIND=dp), DIMENSION(:, :), POINTER            :: rwork_list
```

Comment:
Is it possible to replace these POINTERs with ALLOCATABLEs?

Reply:
I have done so where possible now. In some cases such as for ```glob_loc_list```, ```glob_cell_v```, ```glob_loc_list_a```, ```unique_list_a```, I have not changed to ALLOCATABLEs as they are defined above in `src/manybody_potential.F``` and used throughout in different routines and modules.